File: crates/flow/tests/test_data/sample.rs
// SPDX-FileCopyrightText: 2025 Knitli Inc. <knitli@knit.li>
// SPDX-License-Identifier: AGPL-3.0-or-later

//! Sample Rust code for testing ThreadParse functionality

use std::collections::HashMap;
use std::path::PathBuf;
use serde::{Deserialize, Serialize};

/// A sample struct representing a user
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct User {
    pub id: u64,
    pub name: String,
    pub email: String,
}

/// A sample enum for user roles
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum Role {
    Admin,
    User,
    Guest,
}

/// Process user data and return a result
pub fn process_user(user: &User) -> Result<String, String> {
    if user.name.is_empty() {
        return Err("Name cannot be empty".to_string());
    }

    let formatted = format!("User: {} ({})", user.name, user.email);
    Ok(formatted)
}

/// Calculate total from a list of values
pub fn calculate_total(values: &[i32]) -> i32 {
    values.iter().sum()
}

/// Main function with multiple calls
pub fn main() {
    let user = User {
        id: 1,
        name: "Alice".to_string(),
        email: "alice@example.com".to_string(),
    };

    match process_user(&user) {
        Ok(result) => println!("{}", result),
        Err(e) => eprintln!("Error: {}", e),
    }

    let numbers = vec![1, 2, 3, 4, 5];
    let total = calculate_total(&numbers);
    println!("Total: {}", total);
}

File: crates/flow/tests/test_data/sample.py
#!/usr/bin/env python3

# SPDX-FileCopyrightText: 2025 Knitli Inc. <knitli@knit.li>
# SPDX-FileCopyrightText: 2026 Knitli Inc.
#
# SPDX-License-Identifier: AGPL-3.0-or-later

"""Sample Python code for testing ThreadParse functionality"""

import os
import sys
from typing import List, Dict, Optional
from dataclasses import dataclass


@dataclass
class User:
    """Represents a user in the system"""
    id: int
    name: str
    email: str


class UserManager:
    """Manages user operations"""

    def __init__(self):
        self.users: Dict[int, User] = {}

    def add_user(self, user: User) -> None:
        """Add a user to the manager"""
        if not user.name:
            raise ValueError("Name cannot be empty")
        self.users[user.id] = user

    def get_user(self, user_id: int) -> Optional[User]:
        """Retrieve a user by ID"""
        return self.users.get(user_id)

    def calculate_total(self, values: List[int]) -> int:
        """Calculate sum of values"""
        return sum(values)


def process_user(user: User) -> str:
    """Process user data and return formatted string"""
    if not user.name:
        raise ValueError("Name cannot be empty")
    return f"User: {user.name} ({user.email})"


def main():
    """Main entry point"""
    user = User(id=1, name="Alice", email="alice@example.com")
    manager = UserManager()

    manager.add_user(user)
    result = process_user(user)
    print(result)

    numbers = [1, 2, 3, 4, 5]
    total = manager.calculate_total(numbers)
    print(f"Total: {total}")


if __name__ == "__main__":
    main()
